File: Android/java/doraemonkit-weex/src/main/java/com/didichuxing/doraemonkit/weex/storage/StorageAdapter.java
package com.didichuxing.doraemonkit.weex.storage;

import android.content.Context;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.didichuxing.doraemonkit.widget.recyclerview.AbsRecyclerAdapter;
import com.didichuxing.doraemonkit.widget.recyclerview.AbsViewBinder;
import com.didichuxing.doraemonkit.weex.R;

/**
 * @author haojianglong
 * @date 2019-06-18
 */

public class StorageAdapter extends AbsRecyclerAdapter<StorageAdapter.ViewHolder, StorageInfo> {

    private OnItemClickListener onItemClickListener;
    private final String KEY = "key";
    private final String VALUE = "value";

    private final int TITLE_BACKGROUND_COLOR = Color.parseColor("#BBBBBB");
    private final int NORMAL_BACKGROUND_COLOR = Color.parseColor("#CDCDCD");

    public StorageAdapter(Context context) {
        super(context);
        append(new StorageInfo(KEY, VALUE));
    }

    @Override
    protected ViewHolder createViewHolder(View view, int viewType) {
        return new ViewHolder(view);
    }

    @Override
    protected View createView(LayoutInflater inflater, ViewGroup parent, int viewType) {
        return inflater.inflate(R.layout.dk_item_storage_watch, parent, false);
    }

    public class ViewHolder extends AbsViewBinder<StorageInfo> {

        private TextView key;
        private TextView value;

        public ViewHolder(View view) {
            super(view);
        }

        @Override
        protected void getViews() {
            key = getView(R.id.tv_tip_key);
            value = getView(R.id.tv_tip_value);
        }

        @Override
        public void bind(final StorageInfo bean) {
            if (getAdapterPosition() == 0) {
                key.setBackgroundColor(TITLE_BACKGROUND_COLOR);
                value.setBackgroundColor(TITLE_BACKGROUND_COLOR);
            } else {
                key.setBackgroundColor(NORMAL_BACKGROUND_COLOR);
                value.setBackgroundColor(NORMAL_BACKGROUND_COLOR);
            }
            key.setText(bean.key);
            value.setText(bean.value);
            getView().setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (getAdapterPosition() != 0 && onItemClickListener != null) {
                        onItemClickListener.onItemClick(bean);
                    }
                }
            });
        }

    }

    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
        this.onItemClickListener = onItemClickListener;
    }

    public interface OnItemClickListener {
        /**
         * 点击事件
         *
         * @param info
         */
        void onItemClick(StorageInfo info);
    }

}

File: Android/kotlin/doraemonkit-weex/src/main/java/com/didichuxing/doraemonkit/weex/storage/adapter/StorageAdapter.kt
package com.didichuxing.doraemonkit.weex.storage.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.didichuxing.doraemonkit.weex.R
import com.didichuxing.doraemonkit.weex.storage.StorageInfo
import com.didichuxing.doraemonkit.widget.recyclerview.AbsRecyclerAdapter
import com.didichuxing.doraemonkit.widget.recyclerview.AbsViewBinder

/**
 * Transformed by alvince on 2020/7/1
 *
 * @author haojianglong
 * @date 2019-06-18
 */
class StorageAdapter(context: Context) : AbsRecyclerAdapter<StorageAdapter.ViewHolder, StorageInfo>(context) {

    interface OnItemClickListener {
        /** 点击事件 */
        fun onItemClick(info: StorageInfo)
    }

    companion object {
        private const val TITLE_BACKGROUND_COLOR: Int = 0xFFBBBBBB.toInt()
        private const val NORMAL_BACKGROUND_COLOR: Int = 0xFFCDCDCD.toInt()

        private const val PRE_HOLDER_KEY: String = "key"
        private const val PRE_HOLDER_VALUE = "value"
    }

    var onItemClickListener: OnItemClickListener? = null

    init {
        append(StorageInfo(PRE_HOLDER_KEY, PRE_HOLDER_VALUE))
    }

    override fun createViewHolder(view: View, viewType: Int): ViewHolder =
        ViewHolder(view) {
            onItemClickListener?.onItemClick(it)
        }

    override fun createView(inflater: LayoutInflater, parent: ViewGroup?, viewType: Int): View =
        inflater.inflate(R.layout.dk_item_storage_watch, parent, false)


    class ViewHolder(itemView: View, private val onHolderSelect: (StorageInfo) -> Unit) :
        AbsViewBinder<StorageInfo>(itemView) {

        private val keyTextView: TextView?
            get() = itemView.findViewById(R.id.tv_tip_key)

        private val valueTextView: TextView?
            get() = itemView.findViewById(R.id.tv_tip_value)

        override fun onBind(data: StorageInfo, position: Int) {
            if (adapterPosition == 0) {
                TITLE_BACKGROUND_COLOR
            } else {
                NORMAL_BACKGROUND_COLOR
            }.also { color ->
                keyTextView?.setBackgroundColor(color)
                valueTextView?.setBackgroundColor(color)
            }

            keyTextView?.text = data.key
            valueTextView?.text = data.value

            itemView.setOnClickListener {
                data.takeIf { adapterPosition != 0 }
                    ?.also {
                        onHolderSelect.invoke(it)
                    }
            }
        }
    }

}